dashboard/client/src/common/MultiTabLogViewer.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Box,
+  createStyles,
+  IconButton,
+  makeStyles,
+  Tab,
+  Tabs,
+  Typography,
+} from "@material-ui/core";
+import React, { useState } from "react";
+import { RiArrowUpDownLine } from "react-icons/ri";
+import { useStateApiLogs } from "../pages/log/hooks";
+import { LogViewer } from "../pages/log/LogViewer";
+import { HideableBlock } from "./CollapsibleSection";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    tabs: {
+      borderBottom: `1px solid ${theme.palette.divider}`,
+    },
+  }),
+);
+
+export type MultiTabLogViewerTab = {
+  title: string;
+  nodeId: string | null;
```

Comment:
nit: is it possible to conform optional fields as undefined? i think this field for example is read from the state API - is it b/c `axios.get` sets that field to null?

Reply:
Unfortunately not. The API response returns null sometimes. There is also a place where we need to determine between undefined (variable not passed in) vs null